test/fixtures/dialects/postgres/postgres_shorthand_cast_in_values.sql

```diff
@@ -0,0 +1,7 @@
+WITH t (col_1, col_2) AS (
+    VALUES
+    ('08RIX0', 0.435::NUMERIC(4, 3))
+)
```

Comment:
Prior to using `ExpressionSegment` `VALUES` couldn't handle shorthand casts. Now we get it for free 😎 
[source](https://github.com/sqlfluff/sqlfluff/issues/2362#issuecomment-1018549297)